rotkehlchen/exchanges/coinbase.py

```diff
@@ -160,21 +161,117 @@ def __init__(
             secret=secret,
             database=database,
         )
+        self.is_legacy_api_key=self.determine_key_type(api_key)
         self.apiversion = 'v2'
         self.base_uri = 'https://api.coinbase.com'
         self.msg_aggregator = msg_aggregator
-        self.session.headers.update({'CB-ACCESS-KEY': self.api_key})
-        self.session.headers.update({'CB-VERSION': CB_VERSION})
+        self.host = 'api.coinbase.com'
+
+        
+        if self.is_legacy_api_key:
+            self.session.headers.update({'CB-ACCESS-KEY': self.api_key})
+            self.session.headers.update({'CB-VERSION': CB_VERSION})
+
+    def determine_key_type(self, api_key)-> bool:
+        """
+        Determines if the provided API key and secret are legacy or new.
+        Returns True if legacy, False if new.
+        """
+        legacy_pattern = r'^[\w]+$' 
+        new_pattern = r'^organizations/[\w-]+/apiKeys/[\w-]+$'
```

Comment:
Both of these. `legacy_pattern` and `new_pattern` are constants. Define them at the top of the file and not have them always redefined every time this function is called.

In fact, you can do even better. Since it's a regex pattern you can define and compile it as a constant pattern.

```
LEGACY_RE: Final = re.compile(r'^[\w]+$')
NEW_RE: Final = re.compile(r'^organizations/[\w-]+/apiKeys/[\w-]+$')
```

Move those in the top of the file, and have them defined once. Then use them in the function below like:

`LEGACY_RE.match(api_key)` etc.